### DISCUSSION TITLE
Talos on Fujitsu Futro S920: CPUs are running in "powersaving mode" (998MHz instead of 2.2GHz)

### DISCUSSION THREAD

**Discussion by johanneskastl** [siderolabs/talos#9194] [Q&A]
Hi all,

I think I have found the reason why Talos on my small Fujitsu Futro S920 is having so many problems:

The CPU has 2 cores with a maximum of 2.2GHz. As is usual, the CPU frequency can be increased or decreased depending on the actual load.

Having a look in the Talos Dashboard I can see the frequency being stuck at 998MHz. So the CPU runs in powersaving mode and does not properly adapt the frequency.

I needed multiple installs to get this working, as I needed extensions for Longhorn and tried to get secure boot running. During those multiple tests I have seen the frequency shown as 2.2GHz and also seen it change from 2.2GHz to other values. So Talos (or the linux kernel) is capable of properly adapting the frequency.

I am currently running Talos 1.7.5 with Kubernetes 1.30.3, I will upgrade to 1.7.6 and see if this improves/solves the issue.

Any ideas how I can debug why this CPU frequency adaption is no longer working?

Kind Regards
Johannes

**Comment by johanneskastl** [comment]
Fun fact: As soon as I issued the `talosctl upgrade ...` command, the CPU frequency started jumping from 998MHz to 1.2GHz or 2.2GHz. Let's see what happens after the ugprade is through.

**Comment by smira** [comment]
You probably need to look into [CPU scaling governor](https://wiki.archlinux.org/title/CPU_frequency_scaling#Scaling_governors). The sysfs setting is available in the Talos machine configuration.
